### PR TITLE
paraview: Fix configure arguments for version >= 5.8

### LINKED ISSUE #19135: Installation issue: Paraview - Needs Updated configure args for @5.8.0:
From @chuckatkins, The `package.py` for `paraview` unconditionally sets a number of `configure_args` for paraview to use various dependencies from other Spack packages (e.g.: `USE_SYSTEM_...`).  These arguments have been change in paraview since version 5.8.0 and need an appropriate code path in Spack's `package.py` file.

/cc @danlipsa

### PR DESCRIPTION
**Pull request by chuckatkins**
Fixes #19135

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
@danlipsa may also want to review

File: var/spack/repos/builtin/packages/paraview/package.py [hunk 1]
```diff
@@ -218,24 +218,32 @@ def nvariant_bool(feature):
...
-            '-DVTK_USE_SYSTEM_ZLIB:BOOL=ON',
-            '-DVTK_USE_SYSTEM_PNG:BOOL=ON',
-            '-DOpenGL_GL_PREFERENCE:STRING=LEGACY'
-        ]
+            '-DOpenGL_GL_PREFERENCE:STRING=LEGACY']
+
+        if spec.satisfies('@5.8:'):
+            cmake_args.extend([
+                '-DPARAVIEW_USE_QT:BOOL=%s' % variant_bool('+qt'),
+                '-DVTK_USE_EXTERNAL=ON',
```
Comment (danlipsa) [inline var/spack/repos/builtin/packages/paraview/package.py]: is this right? it does not seem to be in master.

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/paraview/package.py]: It's a bit off.  It should be `-DPARAVIEW_BUILD_WITH_EXTERNAL=ON`.  I'm testing it now, although it reveals a few missing dependencies that need addressing.  I'm moving this to draft until I can finish vetting it.

**Comment by chuckatkins** [comment]
@opadron @danlipsa Okay, this should be good now.  It ended up being much more extensive than initially anticipated.  Tested with 5.6.2, 5.7.0, and 5.8.1 to hit all three flavors of arguments.

**Comment by chuckatkins** [comment]
@opadron can you confirm that this fixes the CI build issue you're hitting?

**Comment by danlipsa** [comment]
LGTM +1

**Pull request merged**